html2text.go

```diff
@@ -27,14 +21,56 @@ type textifyTraverseCtx struct {
 	justClosedDiv   bool
 }
 
+func trimSpace(in string) string {
+	res := []string{}
+	current := []byte{}
+	for _, v := range []byte(in) {
+		switch v {
+		case ' ', '\t', '\n', '\r':
+			if len(current) > 0 {
+				res = append(res, string(current))
+				current = []byte{}
+			}
+		default:
+			current = append(current, v)
+		}
+	}
+	if len(current) > 0 {
+		// last word
+		res = append(res, string(current))
+	}
+	return strings.Join(res, " ")
+}
```

Comment:
I would consider using `strings.FieldsFunc` (used in `strings.Fields`) as far as we split the string.